src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -1309,11 +1310,28 @@ NEVER_INLINE Object* ByteCodeInterpreter::newOperation(ExecutionState& state, co
     return nullptr;
 }
 
+// http://www.ecma-international.org/ecma-262/6.0/index.html#sec-instanceofoperator
 NEVER_INLINE Value ByteCodeInterpreter::instanceOfOperation(ExecutionState& state, const Value& left, const Value& right)
 {
-    if (!right.isFunction()) {
+#if ESCARGOT_ENABLE_ES2015
+    // If Type(C) is not Object, throw a TypeError exception.
+    if (!right.isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, errorMessage_InstanceOf_NotFunction);
+    }
+    // Let instOfHandler be GetMethod(C,@@hasInstance).
+    Value instOfHandler = Object::getMethod(state, right, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().hasInstance));
+    // If instOfHandler is not undefined, then
+    if (!instOfHandler.isUndefined()) {
+        // Return ToBoolean(Call(instOfHandler, C, «O»)).
+        Value arg[1] = { left };
+        return Value(FunctionObject::call(state, instOfHandler, right, 1, arg).toBoolean(state));
```

Comment:
I think `&left` can be used here so the `arg` array is unnecessary.

Reply:
@rerobika Good suggestion! However to use `&left` instead of `arg` array at here, the pointer of `left` should be translated to `Value*` (const Value* -> Value*) since `left` is a const type reference. Furthermore, the `left` value comes from caller's register file [(interpreter #1081)](https://github.com/pando-project/escargot/blob/d9d152dfa28e1fe69752db91b54f7161628120bd/src/interpreter/ByteCodeInterpreter.cpp#L1081). So, I think just allocating an array and copying its value to this array is more suitable and safe than directly accessing the caller's stack address.